src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -87,6 +91,20 @@ public function getRelationFieldsWithPivot()
         });
     }
 
+    /**
+     * Get all fields with 1-n relation set (repeatable is true).
+     *
+     * @return array The fields with 1-n relationships.
+     */
+    public function getRelationFieldsWithRepeatable()
+    {
+        $all_relation_fields = $this->getRelationFields();
+
+        return Arr::where($all_relation_fields, function ($value, $key) {
+            return isset($value['type']) && $value['type'] && $value['type'] == 'repeatable';
```

Comment:
We shouldn't have a mention of the `repeatable` field in CrudPanel traits; this layer should not work depending on field **types**, but on field **attributes** (because elsewhere in the code, attributes define functionality, not field name). This way, we would not condition this new functionality to _just_ the `repeatable` field type, but any other field that wants to implement the functionality. For example:
- a forked field that does something _a little_ different;
- a completely new field that does _everything_ different;

By NOT doing things dependant on field **type**, but field **attributes**, we allow any other field to also benefit from this functionality, without having to add them in this CrudPanel trait (which would be difficult or impossible for custom fields).

At first look, I think we can add an attribute to `repeatable` fields that have relationships, that tells CrudPanel that the field contains a related n-n relationship that _should_ be saved. I think a good start would be a `save_relationship` attribute. Which upon `addField()` would default to:
- `true` for 1-1 and 1-n relationships;
- `false` for n-n relationships;

Then inside the `repeatable` field, to achieve the same functionality we'd just tell the developer to specify `'save_relationship' => true` to overwrite that default.

So the code above, inside the `create()` and `update()` methods, would turn into something like:
```diff
-        $nn_relationships = Arr::pluck($this->getRelationFieldsWithPivot(), 'name');
+        $nn_relationships = Arr::pluck($this->getRelationFieldsThatShouldntBeSaved(), 'name');
        $item = $this->model->create(Arr::except($data, $nn_relationships));
```
Of course, we'll probably need to rename `$nn_relationships` to something like `$relationships_to_not_save`.

Reply:
I am working on this, by having the repeatable field use the same name of the relationship we will infer the relationship info from there. 

So might not need any aditional flag here.

Reply:
Great! Not a priority, though @pxpm - this is tagged to be done in 4.2, it's a feature, not a bug. I've made the mistake of getting distracted and worked on this - don't make the same mistake! 🤣